pkg/service/roommanager.go

```diff
@@ -616,6 +622,70 @@ func (r *RoomManager) handleRTCMessage(ctx context.Context, roomName livekit.Roo
 		pLogger.Debugw("updating room")
 		room.SetMetadata(rm.UpdateRoomMetadata.Metadata)
 	}
+	return nil
+}
+
+func (r *RoomManager) RelaySignal(stream psrpc.ServerStream[*rpc.RelaySignalResponse, *rpc.RelaySignalRequest]) (err error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	req, ok := <-stream.Channel()
+	if !ok {
+		return nil
+	}
+
+	if req.StartSession == nil {
+		return errors.New("expected start session message")
+	}
+
+	prometheus.IncrementParticipantRtcInit(1)
+
+	pi, err := routing.ParticipantInitFromStartSession(req.StartSession, r.config.Region)
+	if err != nil {
+		return errors.Wrap(err, "failed to read participant from session")
+	}
+
+	reqChan := routing.NewDefaultMessageChannel()
+	resChan := routing.NewDefaultMessageChannel()
+	defer func() {
+		reqChan.Close()
+		resChan.Close()
+	}()
+
+	err = r.StartSession(ctx, livekit.RoomName(req.StartSession.RoomName), *pi, reqChan, resChan)
+	if err != nil {
+		logger.Errorw("could not handle new participant", err,
+			"room", req.StartSession.RoomName,
+			"participant", req.StartSession.Identity,
+		)
+	}
+
+	for {
+		select {
+		case msg, ok := <-stream.Channel():
+			if !ok {
```

Comment:
when this happens, does it mean the connection has severed? if so it'd be great to log here.